src/Microsoft.AspNetCore.Mvc.Core/Internal/BindPropertyAttributeApplicationModelProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    internal class BindPropertyAttributeApplicationModelProvider : IApplicationModelProvider
```

Comment:
Generally we'd try to do this in the default provider. Is there a really good reason why this is separate?

Reply:
I like the idea of discovery happening in stages, but that's the not the existing design. I don't want to do 10% of it differently.

Reply:
Not a particularly good reason. I just wanted to treat it like some of the other property attribute providers we have like `TempData` \ `ViewData`.

Reply:
That's how it's done for pages too. That said, we basically allow any binding specific metadata on Razor Page handler to apply to the property:

https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/DefaultPageApplicationModelProvider.cs#L268-L276

IMO, specifically looking for `BindPropertyAttribute` seems like a safer \ better thing to do than indiscriminately pick metadata from the controller.